apps/src/turtle/customLevelBlocks.js

```diff
@@ -39,28 +39,80 @@ function createACircleCode (size, gensym, indent) {
     indent + '}\n'].join('\n');
 }
 
-
 /**
- * Same as draw_a_square, except inputs are not inlined
+ * Returns an initialization object that sets up blockly attributes;
+ *
+ * @param title - The title of the block that will be visible to the user.
+ * @param [argument] - Optional parameter for blocks that accept a value
+ *    argument. This is the title of the argument.
+ * @param [titleValue] - Optional parameter to associate a value with the
+ *    title.
+ * @return the initialization object
  */
-function installDrawASquare(blockly, generator, gensym) {
-  // Create a fake "draw a square" function so it can be made available to users
-  // without being shown in the workspace.
-  blockly.Blocks.draw_a_square_custom = {
-    // Draw a square.
+function makeInitializer(title, argument, titleValue) {
+  return {
     init: function() {
       this.setHSV(94, 0.84, 0.60);
-      this.appendDummyInput()
-          .appendTitle(msg.drawASquare());
-      this.appendValueInput('VALUE')
-          .setAlign(blockly.ALIGN_RIGHT)
-          .setCheck(Blockly.BlockValueType.NUMBER)
-              .appendTitle(msg.lengthParameter() + ':');
+
+      this.appendDummyInput().appendTitle(title, titleValue);
+
+      if (argument !== undefined) {
+        this.appendValueInput('VALUE')
+            .setAlign(Blockly.ALIGN_RIGHT)
+            .setCheck(Blockly.BlockValueType.NUMBER)
+            .appendTitle(argument + ':');
+      }
+
       this.setPreviousStatement(true);
       this.setNextStatement(true);
       this.setTooltip('');
     }
   };
+}
```

Comment:
I'm pretty happy with the general approach, but wonder if we can do better on the naming. Reading this, it's hard to know exactly what title/argument/titleValue mean. Part of this is because the underlying Blockly APIs aren't super clear themselves.

First off, I'd maybe call this `makeBlockInitializer` or even just `makeBlock` instead of `makeInitializer`. Essentially what  we're creating here is our block definition. `makeBlock` might not be be quite accurate, since we haven't yet created a Blockly Block, so I think I might like `makeBlockIniitializer` best.

I also think that the extra level of indirection with `makeInitializerWithAssociatedTitleValue`, etc. might just make things harder to follow. I'd just stick with calling makeInitializer directly everywhere.

`titleValue` is really more of a `titleKey` I think? It's what allows us to later extract the value of title when we call ` var type = this.getTitleValue('TYPE');`

I might call `argument` `parameter` instead, mostly because it seems like that's how we generally refer to it elsewhere (i.e. msg.sizeParameter())